staging/src/k8s.io/component-base/metrics/timing_histogram_test.go

```diff
@@ -0,0 +1,371 @@
+
+package metrics
+
+import (
+	"testing"
+	"time"
+
+	"github.com/blang/semver/v4"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	testclock "k8s.io/utils/clock/testing"
+)
+
+func TestTimingHistogram(t *testing.T) {
+	helpPrefix := "EXPERIMENTAL: "
+	v115 := semver.MustParse("1.15.0")
+	var tests = []struct {
+		desc string
+		*TimingHistogramOpts
+		registryVersion     *semver.Version
+		expectedMetricCount int
+		expectedHelp        string
+	}{
+		{
+			desc: "Test non deprecated",
+			TimingHistogramOpts: &TimingHistogramOpts{
+				Namespace:    "namespace",
+				Name:         "metric_test_name",
+				Subsystem:    "subsystem",
+				Help:         "histogram help message",
+				Buckets:      DefBuckets,
+				InitialValue: 13,
+			},
+			registryVersion:     &v115,
+			expectedMetricCount: 1,
+			expectedHelp:        "[ALPHA] histogram help message",
+		},
+		{
+			desc: "Test deprecated",
+			TimingHistogramOpts: &TimingHistogramOpts{
+				Namespace:         "namespace",
+				Name:              "metric_test_name",
+				Subsystem:         "subsystem",
+				Help:              "histogram help message",
+				DeprecatedVersion: "1.15.0",
+				Buckets:           DefBuckets,
+				InitialValue:      3,
+			},
+			registryVersion:     &v115,
+			expectedMetricCount: 1,
+			expectedHelp:        "[ALPHA] (Deprecated since 1.15.0) histogram help message",
+		},
+		{
+			desc: "Test hidden",
+			TimingHistogramOpts: &TimingHistogramOpts{
+				Namespace:         "namespace",
+				Name:              "metric_test_name",
+				Subsystem:         "subsystem",
+				Help:              "histogram help message",
+				DeprecatedVersion: "1.14.0",
+				Buckets:           DefBuckets,
+				InitialValue:      5,
+			},
+			registryVersion:     &v115,
+			expectedMetricCount: 0,
+			expectedHelp:        "histogram help message",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			registry := newKubeRegistry(apimachineryversion.Info{
+				Major:      "1",
+				Minor:      "15",
+				GitVersion: "v1.15.0-alpha-1.12345",
+			})
+			t0 := time.Now()
+			clk := testclock.NewFakePassiveClock(t0)
+			c := NewTestableTimingHistogram(clk.Now, test.TimingHistogramOpts)
+			registry.MustRegister(c)
+
+			metricChan := make(chan prometheus.Metric)
+			go func() {
+				c.Collect(metricChan)
+				close(metricChan)
+			}()
+			m1 := <-metricChan
+			gm1, ok := m1.(GaugeMetric)
+			if !ok || gm1 != c.GaugeMetric {
+				t.Error("Unexpected metric", m1, c.GaugeMetric)
+			}
+			m2, ok := <-metricChan
+			if ok {
+				t.Error("Unexpected second metric", m2)
+			}
+
+			ms, err := registry.Gather()
+			assert.Equalf(t, test.expectedMetricCount, len(ms), "Got %v metrics, Want: %v metrics", len(ms), test.expectedMetricCount)
+			assert.Nil(t, err, "Gather failed %v", err)
+
+			for _, metric := range ms {
+				expectedHelp := helpPrefix + test.expectedHelp
```

Comment:
Instead of playing with setting expectedHelp here, please set the expectedHelp field in all structs appropriate to have the EXPERIMENAL: prefix.

[Same for vector test.]